### PR TITLE
Dump network to string

### PR DESCRIPTION
**Pull request by yichen88**
Signed-off-by: yichen88 <tang.yichenyves@gmail.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: java/src/main/java/com/powsybl/python/PyPowsyblApiLib.java [hunk 1]
```diff
@@ -291,6 +291,35 @@ public static void dumpNetwork(IsolateThread thread, ObjectHandle networkHandle,
...
+    public static CCharPointer dumpNetworkToString(IsolateThread thread, ObjectHandle networkHandle, CCharPointer format,
+                                                   CCharPointerPointer parameterNamesPtrPtr, int parameterNamesCount,
+                                                   CCharPointerPointer parameterValuesPtrPtr, int parameterValuesCount,
+                                                   ExceptionHandlerPointer exceptionHandlerPtr) {
+        return doCatch(exceptionHandlerPtr, () -> {
+            Network network = ObjectHandles.getGlobal().get(networkHandle);
+            String formatStr = CTypeUtil.toString(format);
+            Properties parameters = createParameters(parameterNamesPtrPtr, parameterNamesCount, parameterValuesPtrPtr, parameterValuesCount);
+            MemDataSource dataSource = new MemDataSource();
+            var exporter = Exporters.getExporter(formatStr);
```
Comment (geofjamg) [inline java/src/main/java/com/powsybl/python/PyPowsyblApiLib.java:304]: Check exporter exist

File: java/src/main/java/com/powsybl/python/PyPowsyblApiLib.java [hunk 2]
```diff
@@ -291,6 +291,35 @@ public static void dumpNetwork(IsolateThread thread, ObjectHandle networkHandle,
...
+            exporter.export(network, parameters, dataSource);
+            try {
+                var exts = dataSource.listNames(".*?");
+                if (exts.size() != 1) {
+                    throw new PowsyblException("Datasource file error");
```
Comment (geofjamg) [inline java/src/main/java/com/powsybl/python/PyPowsyblApiLib.java]: This is not really an error but a non handled case. We probably can do it later (not easy?) we should add a FIXME to explain it.

File: pypowsybl/network.py [hunk 3]
```diff
@@ -59,6 +59,16 @@ def dump(self, file: str, format: str = 'XIIDM', parameters: dict = {}):
         """
         _pypowsybl.dump_network(self.ptr, file, format, parameters)
 
+    def dump_to_string(self, format: str = 'XIIDM', parameters: dict = {}):
+        """Save a network to a file using a specified format.
```
Comment (geofjamg) [inline pypowsybl/network.py]: a string

**Review by geofjamg — CHANGES_REQUESTED** [review]

File: java/src/main/java/com/powsybl/python/PyPowsyblApiLib.java (hunk 2 above)
Comment (geofjamg) [reply java/src/main/java/com/powsybl/python/PyPowsyblApiLib.java]: et also a clear exception message that explain that will only support string export for single file format

File: java/src/main/java/com/powsybl/python/PyPowsyblApiLib.java [hunk 4]
```diff
@@ -291,6 +291,35 @@ public static void dumpNetwork(IsolateThread thread, ObjectHandle networkHandle,
...
+                }
+                String ext = Iterables.getOnlyElement(exts).substring(1);
+                try (InputStream is = new ByteArrayInputStream(dataSource.getData(null, ext));
```
Comment (sylvlecl) [inline java/src/main/java/com/powsybl/python/PyPowsyblApiLib.java]: I think you can use straight the method which takes the full name as input

File: cpp/src/bindings.cpp [hunk 5]
```diff
@@ -99,6 +99,9 @@ PYBIND11_MODULE(_pypowsybl, m) {
     m.def("dump_network", &pypowsybl::dumpNetwork, "Dump network to a file in a given format", py::arg("file"),
           py::arg("format"), py::arg("parameters"), py::arg("parameters"));
 
+    m.def("dump_network_to_string", &pypowsybl::dumpNetworkToString, "Dump network in a given format",
+          py::arg("format"), py::arg("parameters"), py::arg("parameters"));
```
Comment (sylvlecl) [inline cpp/src/bindings.cpp]: I think there is one unnecessary "parameters" argument here

File: pypowsybl/network.py [hunk 6]
```diff
@@ -59,6 +59,16 @@ def dump(self, file: str, format: str = 'XIIDM', parameters: dict = {}):
         """
         _pypowsybl.dump_network(self.ptr, file, format, parameters)
 
+    def dump_to_string(self, format: str = 'XIIDM', parameters: dict = {}):
+        """Save a network to a string using a specified format.
+
+        :param format: format to save the network
```
Comment (geofjamg) [inline pypowsybl/network.py]: Could you port it with google style ?

**Review by geofjamg — APPROVED** [review]

**Pull request merged**